src/server/service/kv.rs

```diff
@@ -614,12 +618,17 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
         stream: RequestStream<RaftMessage>,
         sink: ClientStreamingSink<Done>,
     ) {
+        let store_id = self.store_id;
         let ch = self.ch.clone();
         ctx.spawn(async move {
             let res = stream.map_err(Error::from).try_for_each(move |msg| {
                 RAFT_MESSAGE_RECV_COUNTER.inc();
-                let ret = ch.send_raft_msg(msg).map_err(Error::from);
-                future::ready(ret)
+                if msg.get_from_peer().get_store_id() != store_id {
```

Comment:
How about using `ServerChecker`, so every method is checked automatically?

Reply:
Other methods didn't have the store id field, or did you mean to change at the `grpc-rs` side?

Reply:
`ServerChecker` may be a little complex for this case, we can use it when there are more requests need to be verified.